### PR TITLE
initial cut at script to autodoc json models

### LINKED ISSUE #1581: write script to integrate all bokeh.models and their JSON into dev docs
This is to support #1579

### PR DESCRIPTION
**Pull request by bryevdv**
issues: closes #1581

needs to be re-run by hand to replace source/docs/reference/models.rst for now

### CODE REVIEW

File: sphinx/json_ref.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    'bokeh.models.mappers',
+    'bokeh.models.markers',
+    'bokeh.models.plots',
+    'bokeh.models.ranges',
+    'bokeh.models.renderers',
+    'bokeh.models.sources',
+    'bokeh.models.tickers',
+    'bokeh.models.tools',
+    'bokeh.models.widget',
+]
```
Comment (mattpap) [inline sphinx/json_ref.py]: Can we get rid of this ^^? I'm sure no one will remember to update this. You can use the following code:
```
name = "bokeh.models"
ns = vars(importlib.import_module(name))
modules = [ obj for obj in ns.values() if isinstance(obj, types.ModuleType) and obj.__name__.startswith(name) ]
sorted(modules, key=lambda obj: obj.__name__)
```
This includes `bokeh.models.widgets.*`, but that can be easily filtered out.

**Comment by damianavila** [comment]
LGTM, merging...

File: sphinx/json_ref.py (hunk 1 above)
Comment (bryevdv) [inline sphinx/json_ref.py]: Absolutely, in a follow on PR. This is just what I wrote in 10 minutes to get something in the release. I am going to make several issues and master issues today about improvements to docs build.

Comment (mattpap) [inline sphinx/json_ref.py]: OK.

**Pull request merged**